homeassistant/components/xiaomi_miio/fan.py

```diff
@@ -1036,7 +1038,7 @@ def __init__(self, name, device, entry, unique_id):
         elif self._model in [MODEL_AIRHUMIDIFIER_CA4]:
             self._device_features = FEATURE_FLAGS_AIRHUMIDIFIER_CA4
             self._available_attributes = AVAILABLE_ATTRIBUTES_AIRHUMIDIFIER_CA4
-            self._speed_list = [SPEED_LOW, SPEED_MEDIUM, SPEED_HIGH]
+            self._speed_list = [SPEED_AUTO, SPEED_LOW, SPEED_MEDIUM, SPEED_HIGH]
```

Comment:
Auto should not be a speed but a preset mode.

https://developers.home-assistant.io/docs/core/entity/fan#preset-modes

Reply:
Related to https://github.com/home-assistant/core/pull/51791 Wich implements percentage based speeds and `auto` as a preset mode, using the new fan entity model